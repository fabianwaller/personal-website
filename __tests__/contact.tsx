import Contact from "@/app/contact/page";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("Contact", () => {
  it("renders contact page unchanged", () => {
    const component = render(<Contact />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
