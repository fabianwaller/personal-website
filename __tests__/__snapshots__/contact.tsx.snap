// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders contact page unchanged 1`] = `
<DocumentFragment>
  <section
    class="w-full py-8"
    id="contact"
  >
    <div
      class="container relative ml-auto mr-auto w-screen max-w-custom px-8 md:px-0 mb-12"
    >
      <div
        class="text-center"
      >
        <h2>
          Contact
        </h2>
        <span
          class="block text-sm"
        >
          Send me a message.
        </span>
      </div>
    </div>
    <div
      class="container relative ml-auto mr-auto w-screen max-w-custom px-8 md:px-0"
    >
      <form>
        <div
          class="flex flex-col justify-center gap-4 items-stretch"
        >
          <div
            class="flex flex-col space-y-4 md:flex-row md:space-x-2 md:space-y-0"
          >
            <div
              class="w-full space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for=":r0:-form-item"
              >
                First name
              </label>
              <div
                class="group/input rounded-lg p-[2px] transition duration-300"
              >
                <input
                  aria-describedby=":r0:-form-item-description"
                  aria-invalid="false"
                  class="dark:placeholder-text-neutral-600 duration-400 transitionplaceholder:text-neutral-400 flex h-10 w-full rounded-md border border-none border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:border-2 focus-visible:outline-none focus-visible:ring-[2px] focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 shadow-input group-hover/input:shadow-none"
                  id=":r0:-form-item"
                  name="firstname"
                  placeholder="Joel"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="w-full space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for=":r1:-form-item"
              >
                Last name
              </label>
              <div
                class="group/input rounded-lg p-[2px] transition duration-300"
              >
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  class="dark:placeholder-text-neutral-600 duration-400 transitionplaceholder:text-neutral-400 flex h-10 w-full rounded-md border border-none border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:border-2 focus-visible:outline-none focus-visible:ring-[2px] focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 shadow-input group-hover/input:shadow-none"
                  id=":r1:-form-item"
                  name="lastname"
                  placeholder="Miller"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="w-full space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r2:-form-item"
            >
              Email
            </label>
            <div
              class="group/input rounded-lg p-[2px] transition duration-300"
            >
              <input
                aria-describedby=":r2:-form-item-description"
                aria-invalid="false"
                class="dark:placeholder-text-neutral-600 duration-400 transitionplaceholder:text-neutral-400 flex h-10 w-full rounded-md border border-none border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:border-2 focus-visible:outline-none focus-visible:ring-[2px] focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 shadow-input group-hover/input:shadow-none"
                id=":r2:-form-item"
                name="email"
                placeholder="joel@miller.com"
                value=""
              />
            </div>
            <p
              class="text-sm text-muted-foreground"
              id=":r2:-form-item-description"
            >
              I need your email to answer to your message.
            </p>
          </div>
          <div
            class="w-full space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r3:-form-item"
            >
              Message
            </label>
            <div
              class="group/input rounded-lg p-[2px] transition duration-300"
            >
              <textarea
                aria-describedby=":r3:-form-item-description"
                aria-invalid="false"
                class="dark:placeholder-text-neutral-600 duration-400 transitionplaceholder:text-neutral-400 flex min-h-[80px] w-full rounded-md border border-none border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:border-2 focus-visible:outline-none focus-visible:ring-[2px] focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 shadow-input group-hover/input:shadow-none"
                id=":r3:-form-item"
                name="message"
                placeholder="Your message..."
              />
            </div>
          </div>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 bg-secondary text-secondary-foreground hover:bg-secondary/80"
            type="submit"
          >
            Send 
            <svg
              class="lucide lucide-arrow-right"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="m12 5 7 7-7 7"
              />
            </svg>
            <span
              class="absolute inset-x-0 -bottom-px block h-px w-full bg-gradient-to-r from-transparent via-primary to-transparent opacity-0 transition duration-500 group-hover/btn:opacity-100"
            />
            <span
              class="absolute inset-x-10 -bottom-px mx-auto block h-px w-1/2 bg-gradient-to-r from-transparent via-primary-foreground to-transparent opacity-0 blur-sm transition duration-500 group-hover/btn:opacity-100"
            />
          </button>
        </div>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
